src/daft-plan/src/optimization/rules/drop_repartition.rs

```diff
@@ -34,22 +34,7 @@ impl OptimizerRule for DropRepartition {
                 // Repartition1-Repartition2 -> Repartition1
                 plan.with_new_children(&[child_plan.children()[0].clone()])
             }
-            _ => {
-                // Drop a repartition if it would produce the same partition spec as is already produced by the child.
-                let parent_partition_spec = plan.partition_spec();
-                let child_partition_spec = child_plan.partition_spec();
-                if (parent_partition_spec.num_partitions == 1
-                    && child_partition_spec.num_partitions == 1)
-                    || (child_partition_spec == parent_partition_spec
-                        && !matches!(parent_partition_spec.scheme, PartitionScheme::Range))
-                {
-                    // We directly clone the downstream Repartition child rather than creating a new Arc on child_plan to elide
-                    // an extra copy/Arc.
-                    repartition.input.clone()
-                } else {
-                    return Ok(Transformed::No(plan));
-                }
-            }
+            _ => return Ok(Transformed::No(plan)),
```

Comment:
Perhaps worth adding a comment here also that its expected for implicit "dropping" to happen during the logical to physical plan translation, since we don't yet have num_partitions information at this stage.